src/pages/academy/groundControl/subcomponents/GroundControlEditCell.tsx

```diff
@@ -1,103 +1,119 @@
-        <Dialog
-          icon="info-sign"
-          isOpen={this.state.dialogOpen}
-          onClose={this.handleCloseDialog}
-          title="Update Assessment"
-          canOutsideClickClose={true}
-        >
-          <div className={Classes.DIALOG_BODY}>
-            {fieldName} Date: {this.dateInput()}
-          </div>
-          <div className={Classes.DIALOG_FOOTER}>
-            <div className={Classes.DIALOG_FOOTER_ACTIONS}>
-              {controlButton('Confirm Update', IconNames.TICK, this.handleUpdate)}
-              {controlButton('Cancel', IconNames.CROSS, this.handleCloseDialog)}
-            </div>
-          </div>
-        </Dialog>
-      </div>
-    );
-  }
+  const { handleAssessmentChangeDate } = props;
 
-  private dateInput = () => {
-    return (
-      <DateInput
-        formatDate={this.formatDate}
-        onChange={this.handleDateChange}
-        parseDate={this.parseDate}
-        value={this.props.forOpenDate ? this.state.openAt : this.state.closeAt}
-        timePrecision={'minute'}
-        fill={true}
-        maxDate={this.maxDate}
-        closeOnSelection={false}
-      />
-    );
-  };
+  const handleUpdateDate = React.useCallback(() => {
+    if (!newDate) {
+      // Reset date to current date if no date is selected (null date) in the date input
+      showWarningMessage('No date and time selected!', 2000);
+      setNewDate(new Date(currentDateString));
+    } else {
+      const { id, openAt, closeAt } = data;
+      handleAssessmentChangeDate(
+        id,
+        forOpenDate ? newDate.toISOString() : openAt,
+        forOpenDate ? closeAt : newDate.toISOString()
+      );
+      handleCloseDialog();
+    }
+  }, [
+    data,
+    forOpenDate,
+    newDate,
+    currentDateString,
+    handleAssessmentChangeDate,
+    handleCloseDialog
+  ]);
 
-  private parseDate = (str: string) => new Date(str);
+  const handleParseDate = (str: string) => new Date(str);
+  const handleFormatDate = (date: Date) => date.toLocaleString();
 
-  private formatDate = (date: Date) => date.toLocaleString();
+  const handleDateChange = React.useCallback((selectedDate: Date) => setNewDate(selectedDate), []);
+  const handleDateError = React.useCallback(() => {
+    // Reset date to current date if user enters an invalid date string
+    showWarningMessage('Failed to parse date string! Defaulting to current date.', 2000);
+    setNewDate(new Date(currentDateString));
+  }, [currentDateString]);
 
-  private handleDateChange = (selectedDate: Date) => {
-    if (this.props.forOpenDate) {
-      this.setState({ openAt: selectedDate });
-    } else {
-      this.setState({ closeAt: selectedDate });
-    }
-  };
+  const dateInput = (
+    <DateInput
+      formatDate={handleFormatDate}
+      onChange={handleDateChange}
+      onError={handleDateError}
+      parseDate={handleParseDate}
+      placeholder={'MM/DD/YYYY, HH:mm:ss (AM/PM) or select a date'}
```

Comment:
American date format :confused:

Reply:
Unfortunately this is the default JS date format - I'll look for a way to configure this.